### ISSUE TITLE
new node start serving before schema was syncronized and then raft failed to resolve remove node ip

### ISSUE THREAD

**Issue by aleksbykov** [scylladb/scylladb#15686]
## Issue description

- [ ] This issue is a regression.
- [ ] It is unknown if this issue is a regression.

Scylla configured with experimental feature: 
experimental_features:
    - consistent-topology-changes

In case of testing https://github.com/scylladb/scylladb/commit/8576502c486190562035169621aa6fb44ec2df11 in sct was added new nemesis. 
Nemesis choose target node (node5) and stop scylla processes with command
`pkill --signal SIGSTOP -e scylla`
Then after node5 was marked as down:
```
Datacenter: eu-west
===================
Status=Up/Down
|/ State=Normal/Leaving/Joining/Moving
--  Address     Load       Tokens       Owns    Host ID                               Rack
UN  10.4.2.71   1.81 MB    256          ?       08fad1f7-0d0d-4bb4-89c8-1e078b2e097b  1a
UN  10.4.2.82   1.43 MB    256          ?       d80a4f68-13d6-4cb1-b685-05b407e0d984  1a
UN  10.4.0.51   1.27 MB    256          ?       ba680a02-c28b-42e3-b1b9-30b8e10d5652  1a
UN  10.4.2.83   1.36 MB    256          ?       211f4204-7751-49b7-8b03-aae41f9759a7  1a
UN  10.4.3.28   324.07 MB  256          ?       47b04edd-d2e3-435c-9f6a-284c1f33d35a  1a
DN  10.4.2.125  1.67 MB    256          ?       0956fa97-43ad-4c76-bcb8-12e0b1ca300f  1a
```
node5 was removed with `nodetool removenode` successfully
```
Datacenter: eu-west
===================
Status=Up/Down
|/ State=Normal/Leaving/Joining/Moving
--  Address    Load       Tokens       Owns    Host ID                               Rack
UN  10.4.2.71  4.97 GB    256          ?       08fad1f7-0d0d-4bb4-89c8-1e078b2e097b  1a
UN  10.4.2.82  1.44 MB    256          ?       d80a4f68-13d6-4cb1-b685-05b407e0d984  1a
UN  10.4.0.51  1.28 MB    256          ?       ba680a02-c28b-42e3-b1b9-30b8e10d5652  1a
UN  10.4.2.83  1.36 MB    256          ?       211f4204-7751-49b7-8b03-aae41f9759a7  1a
UN  10.4.3.28  324.07 MB  256          ?       47b04edd-d2e3-435c-9f6a-284c1f33d35a  1a
```

Then node5 was terminated, and new node7 had been adding to cluster. But during addint node 7 first try serving before schema was synced,
```
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] messaging_service - Starting Encrypted Messaging Service on SSL port 7001
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] storage_service - entering STARTING mode
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] storage_service - initial_contact_nodes={10.4.3.28, 10.4.2.83, 10.4.2.82, 10.4.2.71, 10.4.0.51}
, loaded_endpoints={}, loaded_peer_features=0
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] storage_service - Performing gossip shadow round
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] gossip - Gossip shadow round started with nodes={10.4.3.28, 10.4.2.83, 10.4.2.82, 10.4.2.71, 10
.4.0.51}
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] gossip - Gossip shadow round finished with nodes_talked={10.4.2.83, 10.4.2.82, 10.4.2.71, 10.4.
0.51, 10.4.3.28}
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] storage_service - Checking bootstrapping/leaving/moving nodes: ok (check_for_endpoint_collision
)
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] storage_service - Save advertised features list in the 'system.local' table
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] schema_tables - Schema version changed to 59adb24e-f3cd-3e02-97f0-5b395827453f
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] storage_service - Starting up server gossip
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] gossip - failure_detector_loop: Started main loop
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] raft_group0 - setup_group0: joining group 0...
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] raft_group0 - server 1d24cdd9-5602-4a5c-ad19-96fccaf5a135 found no local group 0. Discovering..
.
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] raft_group0 - server 1d24cdd9-5602-4a5c-ad19-96fccaf5a135 found group 0 with group id a9c70a70-
66bd-11ee-a82a-b13c3e362405, leader ba680a02-c28b-42e3-b1b9-30b8e10d5652
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] storage_service - raft topology: join: sending the join request to 10.4.0.51
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] storage_service - raft topology: join: request to join placed, waiting for the response from th
e topology coordinator
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] raft_group0 - Server 1d24cdd9-5602-4a5c-ad19-96fccaf5a135 is starting group 0 with id a9c70a70-
66bd-11ee-a82a-b13c3e362405
2023-10-09T16:50:23+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla-jmx[5583]: Picked up JAVA_TOOL_OPTIONS:
2023-10-09T16:50:24+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7   !NOTICE | sudo[5617]: scyllaadm : PWD=/home/scyllaadm ; USER=root ; COMMAND=/usr/bin/cat /etc/scylla.d/io.conf
2023-10-09T16:50:24+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | sudo[5617]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=1000)
2023-10-09T16:50:24+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | sudo[5617]: pam_unix(sudo:session): session closed for user root
2023-10-09T16:50:24+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla-jmx[5583]: Using config file: /etc/scylla/scylla.yaml
2023-10-09T16:50:24+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla-jmx[5583]: Connecting to http://127.0.0.1:10000
2023-10-09T16:50:24+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla-jmx[5583]: Starting the JMX server
2023-10-09T16:50:24+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7   !NOTICE | sudo[5626]: scyllaadm : PWD=/home/scyllaadm ; USER=root ; COMMAND=/usr/bin/curl -o /etc/apt/sources.list.d/scylla-manager.list
 -L http://downloads.scylladb.com.s3.amazonaws.com/deb/ubuntu/scylladb-manager-3.2.list
2023-10-09T16:50:24+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | sudo[5626]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=1000)
2023-10-09T16:50:24+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | sudo[5626]: pam_unix(sudo:session): session closed for user root
2023-10-09T16:50:24+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:goss] gossip - InetAddress 10.4.0.51 is now UP, status = NORMAL
2023-10-09T16:50:24+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:goss] gossip - InetAddress 10.4.2.83 is now UP, status = NORMAL
2023-10-09T16:50:24+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:goss] gossip - InetAddress 10.4.2.82 is now UP, status = NORMAL
2023-10-09T16:50:24+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:goss] gossip - InetAddress 10.4.2.71 is now UP, status = NORMAL
2023-10-09T16:50:24+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:goss] gossip - InetAddress 10.4.3.28 is now UP, status = NORMAL
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla-jmx[5583]: JMX is enabled to receive remote connections on port: 7199
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard 12:stat] storage_proxy - Failed to apply mutation from 10.4.0.51#12: std::_Nested_exception<schema_versi
on_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  1:stat] storage_proxy - Failed to apply mutation from 10.4.0.51#1: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  2:stat] storage_proxy - Failed to apply mutation from 10.4.0.51#2: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  9:stat] storage_proxy - Failed to apply mutation from 10.4.3.28#9: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard 12:stat] storage_proxy - Failed to apply mutation from 10.4.0.51#12: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)

2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  0:stat] storage_proxy - Failed to apply mutation from 10.4.3.28#0: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  9:stat] storage_proxy - Failed to apply mutation from 10.4.3.28#9: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  1:stat] storage_proxy - Failed to apply mutation from 10.4.0.51#1: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard 12:stat] storage_proxy - Failed to apply mutation from 10.4.0.51#12: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  7:stat] storage_proxy - Failed to apply mutation from 10.4.0.51#7: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  5:stat] storage_proxy - Failed to apply mutation from 10.4.0.51#5: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  1:stat] storage_proxy - Failed to apply mutation from 10.4.0.51#1: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard 12:stat] storage_proxy - Failed to apply mutation from 10.4.0.51#12: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  9:stat] storage_proxy - Failed to apply mutation from 10.4.3.28#9: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  7:stat] storage_proxy - Failed to apply mutation from 10.4.0.51#7: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  5:stat] storage_proxy - Failed to apply mutation from 10.4.0.51#5: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  1:stat] storage_proxy - Failed to apply mutation from 10.4.0.51#1: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard 12:stat] storage_proxy - Failed to apply mutation from 10.4.0.51#12: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 824c9691-66c2-11ee-a38d-01af987a8afa): data_dictionary::no_such_keyspace (Can't find a keyspace keyspace1)

``` 

and then bootstrap is stuck, because it could get ip for remove node:
```
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] schema_tables - Schema version changed to b53ac3ba-66c2-11ee-7580-8d82581e14e7
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] schema_tables - Schema version changed to b53ac3ba-66c2-11ee-7580-8d82581e14e7
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] schema_tables - Schema version changed to b53ac3ba-66c2-11ee-7580-8d82581e14e7
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] schema_tables - Schema version changed to b53ac3ba-66c2-11ee-7580-8d82581e14e7
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] schema_tables - Schema version changed to b53ac3ba-66c2-11ee-7580-8d82581e14e7
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] format_selector - Selected me sstables format
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] features - Feature UDA is enabled
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] features - Feature EMPTY_REPLICA_PAGES is enabled
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] features - Feature TOMBSTONE_GC_OPTIONS is enabled
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] features - Feature AGGREGATE_STORAGE_OPTIONS is enabled
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] features - Feature TYPED_ERRORS_IN_READ_RPC is enabled
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] features - Feature SUPPORTS_RAFT_CLUSTER_MANAGEMENT is enabled
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] features - Feature SCHEMA_COMMITLOG is enabled
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:stre] features - Feature UDA_NATIVE_PARALLELIZED_AGGREGATION is enabled
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  0:stre] storage_service - raft topology: cannot map 0956fa97-43ad-4c76-bcb8-12e0b1ca300f to ip, retryin
g.
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:comp] compaction - [Compact system.scylla_local f1bb8bc0-66c3-11ee-9b1a-27648816ab7d] Compacted 2 sst
ables to [/var/lib/scylla/data/system/scylla_local-2972ec7ffb2038ddaac1d876f2e3fcbd/md-3ga2_1as1_360xt2cwu7pi6x9enx-big-Data.db:level=0]. 10kB to 5kB (~53% of original) in 6ms = 1MB/s. ~256 total partitions merg
ed to 3.
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7     !INFO | scylla[5530]:  [shard  0:comp] compaction - [Compact system.scylla_local f1bc9d30-66c3-11ee-9b1a-27648816ab7d] Compacting [/va
r/lib/scylla/data/system/scylla_local-2972ec7ffb2038ddaac1d876f2e3fcbd/md-3ga2_1as1_35li82cwu7pi6x9enx-big-Data.db:level=0:origin=memtable,/var/lib/scylla/data/system/scylla_local-2972ec7ffb2038ddaac1d876f2e3fcb

2023-10-09T16:51:28+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  0:stre] storage_service - (rate limiting dropped 199 similar messages) raft topology: cannot map 0956fa97-43ad-4c76-bcb8-12e0b1ca300f to ip, retrying.
2023-10-09T16:51:29+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  0:stre] storage_service - (rate limiting dropped 199 similar messages) raft topology: cannot map 0956fa97-43ad-4c76-bcb8-12e0b1ca300f to ip, retrying.
2023-10-09T16:51:30+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  0:stre] storage_service - (rate limiting dropped 199 similar messages) raft topology: cannot map 0956fa97-43ad-4c76-bcb8-12e0b1ca300f to ip, retrying.
2023-10-09T16:51:31+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  0:stre] storage_service - (rate limiting dropped 199 similar messages) raft topology: cannot map 0956fa97-43ad-4c76-bcb8-12e0b1ca300f to ip, retrying.
2023-10-09T16:51:32+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  0:stre] storage_service - (rate limiting dropped 199 similar messages) raft topology: cannot map 0956fa97-43ad-4c76-bcb8-12e0b1ca300f to ip, retrying.
2023-10-09T16:51:33+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  0:stre] storage_service - (rate limiting dropped 199 similar messages) raft topology: cannot map 0956fa97-43ad-4c76-bcb8-12e0b1ca300f to ip, retrying.
2023-10-09T16:51:34+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-7  !WARNING | scylla[5530]:  [shard  0:stre] storage_service - (rate limiting dropped 199 similar messages) raft topology: cannot map 0956fa97-43ad-4c76-bcb8-12e0b1ca300f to ip, retrying.

```

## Impact

new node was not bootstrapped 

## How frequently does it reproduce?

*Describe the frequency with how this issue can be reproduced.*

## Installation details

Kernel Version: 5.15.0-1047-aws
Scylla version (or git commit hash): `5.4.0~dev-20231006.498e3ec435be` with build-id `16c6112202348a8adba536b4195d48adfdf958f9`

Cluster size: 6 nodes (i4i.4xlarge)

Scylla Nodes used in this run:
- longevity-100gb-4h-NewNemes-db-node-7c794cca-7 (34.250.33.212 | 10.4.2.230) (shards: -1)
- longevity-100gb-4h-NewNemes-db-node-7c794cca-6 (3.251.67.208 | 10.4.2.71) (shards: 14)
- longevity-100gb-4h-NewNemes-db-node-7c794cca-5 (3.252.248.127 | 10.4.2.125) (shards: 14)
- longevity-100gb-4h-NewNemes-db-node-7c794cca-4 (3.253.31.187 | 10.4.2.82) (shards: 14)
- longevity-100gb-4h-NewNemes-db-node-7c794cca-3 (3.250.19.91 | 10.4.2.83) (shards: 14)
- longevity-100gb-4h-NewNemes-db-node-7c794cca-2 (176.34.91.38 | 10.4.0.51) (shards: 14)
- longevity-100gb-4h-NewNemes-db-node-7c794cca-1 (34.252.184.64 | 10.4.3.28) (shards: 14)

OS / Image: `ami-06f33c2dc88569dd3` (aws: undefined_region)

Test: `longevity-100gb-4h`
Test id: `7c794cca-9ddc-4c04-8240-b9a1d0aa2376`
Test name: `scylla-staging/abykov/longevity-100gb-4h`
Test config file(s):

- [longevity-100gb-4h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/bc3b074d18dc946a2a4905fd3729e7aae85e6550/test-cases/longevity/longevity-100gb-4h.yaml)

<details>
<summary>
Logs and commands
</summary>

- Restore Monitor Stack command: `$ hydra investigate show-monitor 7c794cca-9ddc-4c04-8240-b9a1d0aa2376`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=7c794cca-9ddc-4c04-8240-b9a1d0aa2376)
- Show all stored logs command: `$ hydra investigate show-logs 7c794cca-9ddc-4c04-8240-b9a1d0aa2376`

## Logs:
- **db-cluster-7c794cca.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7c794cca-9ddc-4c04-8240-b9a1d0aa2376/20231009_210649/db-cluster-7c794cca.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7c794cca-9ddc-4c04-8240-b9a1d0aa2376/20231009_210649/db-cluster-7c794cca.tar.gz)
- **sct-runner-events-7c794cca.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7c794cca-9ddc-4c04-8240-b9a1d0aa2376/20231009_210649/sct-runner-events-7c794cca.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7c794cca-9ddc-4c04-8240-b9a1d0aa2376/20231009_210649/sct-runner-events-7c794cca.tar.gz)
- **sct-7c794cca.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7c794cca-9ddc-4c04-8240-b9a1d0aa2376/20231009_210649/sct-7c794cca.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7c794cca-9ddc-4c04-8240-b9a1d0aa2376/20231009_210649/sct-7c794cca.log.tar.gz)
- **loader-set-7c794cca.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7c794cca-9ddc-4c04-8240-b9a1d0aa2376/20231009_210649/loader-set-7c794cca.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7c794cca-9ddc-4c04-8240-b9a1d0aa2376/20231009_210649/loader-set-7c794cca.tar.gz)
- **monitor-set-7c794cca.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7c794cca-9ddc-4c04-8240-b9a1d0aa2376/20231009_210649/monitor-set-7c794cca.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7c794cca-9ddc-4c04-8240-b9a1d0aa2376/20231009_210649/monitor-set-7c794cca.tar.gz)
- **parallel-timelines-report-7c794cca.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7c794cca-9ddc-4c04-8240-b9a1d0aa2376/20231009_210649/parallel-timelines-report-7c794cca.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7c794cca-9ddc-4c04-8240-b9a1d0aa2376/20231009_210649/parallel-timelines-report-7c794cca.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-staging/job/abykov/job/longevity-100gb-4h/63/)
[Argus](https://argus.scylladb.com/test/94b20881-29d4-4c09-909a-d91d2dd43cac/runs?additionalRuns[]=7c794cca-9ddc-4c04-8240-b9a1d0aa2376)
</details>

**Comment by avikivity** [comment]
@kbr-scylla @kostja please evaluate, it looks like a very basic invariant was violated.

**Comment by piodul** [comment]
> > But during addint node 7 first try serving before schema was synced,
> 
> This looks new. Could it be a problem with the new join procedure? cc @piodul

I'm not sure. I looked at the logs and it looks like the new node didn't finish the handshake yet (got stuck on the id2ip translation loop when applying commands during read barrier), so it couldn't even have reached the `bootstrapping` state.

All write errors on the joining node happened at (!) 16:50:25, at the moment when other nodes noticed the new node in gossip:

```
2023-10-09T16:50:25+00:00 longevity-100gb-4h-NewNemes-db-node-7c794cca-6     !INFO | scylla[6019]:  [shard  0:goss] gossip - InetAddress 10.4.2.230 is now UP, status = UNKNOWN
```

Perhaps, in raft mode, we still keep updating token metadata based on gossip notifications in some cases?

**Comment by piodul** [comment]
After a short conversation with @kbr-scylla it turned out that it is an issue with the new procedure for joining nodes. The topology coordinator generates bootstrap tokens and assigns them to the node, __does__ the transition to bootstrap, only then tells the node to perform the read barrier and read the current system topology/schema state.

We need to postpone generating and inserting the tokens only after the node does the read barrier. I'll add it to the umbrella issue about the followups to the join handshake.

**Comment by bhalevy** [comment]
I hit a node serving before schema is synchronized here too:
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-debug/258/testReport/schema_management_test/TestSchemaManagement/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split012___test_update_schema_while_node_is_killed_alter_table_/
```
        case_map = {
            "create_table": create_table_case,
            "alter_table": functools.partial(
                session.execute, "ALTER TABLE ks.cf ADD (c3 text);", timeout=180),
            "drop_table": functools.partial(
                session.execute, "DROP TABLE cf;", timeout=180),
        }
        with ThreadPoolExecutor(max_workers=2) as executor:
            logger.debug(f'2. kill node during {case}')
            kill_node_future = executor.submit(node2.stop, gently=False, wait_other_notice=True)
            case_map[case]()
            kill_node_future.result()
    
        logger.debug("3. Start the stopped node2")
        node2.start(wait_for_binary_proto=True)
    
        session = self.patient_exclusive_cql_connection(node2)
    
        def create_or_alter_table_expected_result(col_mun):
            rows = session.execute(SimpleStatement("SELECT * FROM ks.cf LIMIT 1;",
                                                   consistency_level=ConsistencyLevel.QUORUM))
            assert len(rows_to_list(rows)[0]) == col_mun, \
                f"Expected {col_mun} columns but got rows:{rows} instead"
            for key in range(0, 10):
                query_c1c2(session=session, key=key, consistency=ConsistencyLevel.QUORUM)
    
        expected_case_result_map = {
            "create_table": functools.partial(create_or_alter_table_expected_result, 3),
            "alter_table": functools.partial(create_or_alter_table_expected_result, 4),
            "drop_table": functools.partial(assert_invalid, session, "SELECT * FROM test1"),
        }
        logger.debug('verify that commitlog has been replayed and that all data is restored')
>       expected_case_result_map[case]()

schema_management_test.py:277: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

col_mun = 4

    def create_or_alter_table_expected_result(col_mun):
        rows = session.execute(SimpleStatement("SELECT * FROM ks.cf LIMIT 1;",
                                               consistency_level=ConsistencyLevel.QUORUM))
>       assert len(rows_to_list(rows)[0]) == col_mun, \
            f"Expected {col_mun} columns but got rows:{rows} instead"
E       AssertionError: Expected 4 columns but got rows:<cassandra.cluster.ResultSet object at 0x7f3ba473ce50> instead
E       assert 3 == 4
E        +  where 3 = len(['k1', 'value1', 'value2'])
```

https://jenkins.scylladb.com/job/scylla-master/job/dtest-debug/258/artifact/logs-full.debug.012/dtest-gw1.log
```
07:27:58,855 793     schema_management_test         DEBUG    schema_management_test.py:253  | test_update_schema_while_node_is_killed[alter_table]: 2. kill node during alter_table
07:28:19,410 793     cassandra.cluster              WARNING  libevreactor.py     :370  | test_update_schema_while_node_is_killed[alter_table]: Host 127.0.43.2:9042 has been marked down
07:28:20,718 793     schema_management_test         DEBUG    schema_management_test.py:258  | test_update_schema_while_node_is_killed[alter_table]: 3. Start the stopped node2
07:28:20,724 793     ccm                            DEBUG    cluster.py          :709  | test_update_schema_while_node_is_killed[alter_table]: node2: Starting scylla: args=['/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/bin/scylla', '--options-file', '/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/conf/scylla.yaml', '--log-to-stdout', '1', '--abort-on-seastar-bad-alloc', '--abort-on-lsa-bad-alloc', '1', '--abort-on-internal-error', '1', '--api-address', '127.0.43.2', '--collectd-hostname', 'ip-10-0-4-55.ec2.internal.node2', '--smp', '2', '--memory', '1024M', '--developer-mode', 'true', '--default-log-level', 'info', '--blocked-reactor-notify-ms', '5000', '--collectd', '0', '--overprovisioned', '--prometheus-address', '127.0.43.2', '--unsafe-bypass-fsync', '1', '--kernel-page-cache', '1', '--commitlog-use-o-dsync', '0', '--max-networking-io-control-blocks', '1000'] wait_other_notice=True wait_for_binary_proto=True
07:28:20,727 793     cassandra.pool                 WARNING  thread.py           :58   | test_update_schema_while_node_is_killed[alter_table]: Error attempting to reconnect to 127.0.43.2:9042, scheduling retry in 2.08 seconds: [Errno 111] Tried connecting to [('127.0.43.2', 9042)]. Last error: Connection refused
07:28:22,880 793     cassandra.pool                 WARNING  thread.py           :58   | test_update_schema_while_node_is_killed[alter_table]: Error attempting to reconnect to 127.0.43.2:9042, scheduling retry in 4.0 seconds: [Errno 111] Tried connecting to [('127.0.43.2', 9042)]. Last error: Connection refused
07:28:26,980 793     cassandra.pool                 WARNING  thread.py           :58   | test_update_schema_while_node_is_killed[alter_table]: Error attempting to reconnect to 127.0.43.2:9042, scheduling retry in 3.88 seconds: [Errno 111] Tried connecting to [('127.0.43.2', 9042)]. Last error: Connection refused
07:28:30,961 793     cassandra.pool                 WARNING  thread.py           :58   | test_update_schema_while_node_is_killed[alter_table]: Error attempting to reconnect to 127.0.43.2:9042, scheduling retry in 4.0 seconds: [Errno 111] Tried connecting to [('127.0.43.2', 9042)]. Last error: Connection refused
07:28:35,055 793     cassandra.pool                 WARNING  thread.py           :58   | test_update_schema_while_node_is_killed[alter_table]: Error attempting to reconnect to 127.0.43.2:9042, scheduling retry in 4.0 seconds: [Errno 111] Tried connecting to [('127.0.43.2', 9042)]. Last error: Connection refused
07:28:36,799 793     ccm                            DEBUG    cluster.py          :709  | test_update_schema_while_node_is_killed[alter_table]: node2: Starting scylla-jmx: args=['/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/bin/symlinks/scylla-jmx', '-Dapiaddress=127.0.43.2', '-Djavax.management.builder.initial=com.scylladb.jmx.utils.APIBuilder', '-Djava.rmi.server.hostname=127.0.43.2', '-Dcom.sun.management.jmxremote', '-Dcom.sun.management.jmxremote.host=127.0.43.2', '-Dcom.sun.management.jmxremote.port=7199', '-Dcom.sun.management.jmxremote.rmi.port=7199', '-Dcom.sun.management.jmxremote.local.only=false', '-Xmx256m', '-XX:+UseSerialGC', '-Dcom.sun.management.jmxremote.authenticate=false', '-Dcom.sun.management.jmxremote.ssl=false', '-jar', '/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/bin/scylla-jmx-1.0.jar']
07:28:37,207 793     cassandra.cluster              INFO     thread.py           :58   | test_update_schema_while_node_is_killed[alter_table]: Host 127.0.43.2:9042 may be up; will prepare queries and open connection pool
07:28:37,215 793     cassandra.cluster              INFO     _base.py            :340  | test_update_schema_while_node_is_killed[alter_table]: Connection pools established for node 127.0.43.2:9042
07:28:37,403 793     cassandra.cluster              INFO     dtest_setup.py      :437  | test_update_schema_while_node_is_killed[alter_table]: New Cassandra host <Host: 127.0.43.1:9042 datacenter1> discovered
07:28:37,403 793     cassandra.cluster              INFO     dtest_setup.py      :437  | test_update_schema_while_node_is_killed[alter_table]: New Cassandra host <Host: 127.0.43.3:9042 datacenter1> discovered
07:28:37,670 793     schema_management_test         DEBUG    schema_management_test.py:276  | test_update_schema_while_node_is_killed[alter_table]: verify that commitlog has been replayed and that all data is restored
07:28:38,081 793     errors                         ERROR    conftest.py         :225  | test_update_schema_while_node_is_killed[alter_table]: test failed: 
```

https://jenkins.scylladb.com/job/scylla-master/job/dtest-debug/258/artifact/logs-full.debug.012/1697354954276_schema_management_test.py%3A%3ATestSchemaManagement%3A%3Atest_update_schema_while_node_is_killed%5Balter_table%5D/node1.log
```
INFO  2023-10-15 07:27:58,902 [shard 0:stat] schema_tables - Altering ks.cf id=5cf5a6c0-6b2c-11ee-9e66-5fe7028648ff version=9d351374-8a53-305f-bebe-9221044c1bb9
INFO  2023-10-15 07:27:58,908 [shard 0:stat] query_processor - Column definitions for ks.cf changed, invalidating related prepared statements
INFO  2023-10-15 07:27:58,910 [shard 1:stat] query_processor - Column definitions for ks.cf changed, invalidating related prepared statements
INFO  2023-10-15 07:27:59,151 [shard 0:stat] schema_tables - Schema version changed to 709f0867-6d2f-3135-a41b-4ceb05762e78
```

https://jenkins.scylladb.com/job/scylla-master/job/dtest-debug/258/artifact/logs-full.debug.012/1697354954276_schema_management_test.py%3A%3ATestSchemaManagement%3A%3Atest_update_schema_while_node_is_killed%5Balter_table%5D/node2.log
```
INFO  2023-10-15 07:28:36,162 [shard 0:main] init - Waiting for gossip to settle before accepting client requests...
INFO  2023-10-15 07:28:36,174 [shard 0:main] init - allow replaying hints
INFO  2023-10-15 07:28:36,183 [shard 0:main] init - Launching generate_mv_updates for non system tables
INFO  2023-10-15 07:28:36,192 [shard 0:main] init - starting the view builder
INFO  2023-10-15 07:28:36,345 [shard 0:main] init - starting native transport
WARN  2023-10-15 07:28:36,365 [shard 0:goss] migration_manager - Can't send migration request: node 127.0.43.3 is down.

^^^ unclear why, maybe it's related

INFO  2023-10-15 07:28:36,367 [shard 0:stat] cql_server_controller - Starting listening for CQL clients on 127.0.43.2:19042 (unencrypted, shard-aware)
INFO  2023-10-15 07:28:36,367 [shard 0:stat] cql_server_controller - Starting listening for CQL clients on 127.0.43.2:9042 (unencrypted, non-shard-aware)
INFO  2023-10-15 07:28:36,393 [shard 0:main] init - serving
INFO  2023-10-15 07:28:36,393 [shard 0:main] init - Scylla version 5.4.0~dev-0.20231013.055f0617064d initialization completed.
INFO  2023-10-15 07:28:36,405 [shard 0:comp] compaction - [Compact system_schema.tables 73239970-6b2c-11ee-b854-42bd0f7ab697] Compacted 2 sstables to [/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/data/system_schema/tables-afddfb9dbc1e30688056eed6c302ba09/me-3ga8_0krn_0n5c02swuzk9y7jmaf-big-Data.db:level=0]. 25kB to 14kB (~57% of original) in 1191ms = 21kB/s. ~256 total partitions merged to 7.
INFO  2023-10-15 07:28:36,417 [shard 0:goss] gossip - InetAddress 127.0.43.1 is now UP, status = NORMAL
INFO  2023-10-15 07:28:36,425 [shard 0:goss] migration_manager - Requesting schema pull from 127.0.43.1:0
INFO  2023-10-15 07:28:36,427 [shard 0:goss] migration_manager - Pulling schema from 127.0.43.1:0
INFO  2023-10-15 07:28:36,441 [shard 0:comp] compaction - [Compact system_schema.computed_columns 74048390-6b2c-11ee-b854-42bd0f7ab697] Compacting [/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/data/system_schema/computed_columns-cc7c7069374033c192a4c3de78dbd2c4/me-3ga8_0krk_1yy0g2swuzk9y7jmaf-big-Data.db:level=0:origin=memtable,/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/data/system_schema/computed_columns-cc7c7069374033c192a4c3de78dbd2c4/me-3ga8_0kre_2ig682swuzk9y7jmaf-big-Data.db:level=0:origin=memtable]
INFO  2023-10-15 07:28:36,518 [shard 0:goss] gossip - InetAddress 127.0.43.3 is now UP, status = NORMAL
INFO  2023-10-15 07:28:36,524 [shard 0:goss] migration_manager - Requesting schema pull from 127.0.43.3:0
INFO  2023-10-15 07:28:36,524 [shard 0:goss] migration_manager - Pulling schema from 127.0.43.3:0
INFO  2023-10-15 07:28:36,619 [shard 0:comp] compaction - [Compact system_schema.computed_columns 74048390-6b2c-11ee-b854-42bd0f7ab697] Compacted 2 sstables to [/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/data/system_schema/computed_columns-cc7c7069374033c192a4c3de78dbd2c4/me-3ga8_0kro_35li82swuzk9y7jmaf-big-Data.db:level=0]. 11kB to 6kB (~55% of original) in 154ms = 74kB/s. ~256 total partitions merged to 7.
INFO  2023-10-15 07:28:36,626 [shard 0:comp] compaction - [Compact system_schema.view_virtual_columns 74226bd0-6b2c-11ee-b854-42bd0f7ab697] Compacting [/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/data/system_schema/view_virtual_columns-08843b6345dc3be29798a0418295cfaa/me-3ga8_0krj_3iggw2swuzk9y7jmaf-big-Data.db:level=0:origin=memtable,/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/data/system_schema/view_virtual_columns-08843b6345dc3be29798a0418295cfaa/me-3ga8_0kre_2z60w2swuzk9y7jmaf-big-Data.db:level=0:origin=memtable]
INFO  2023-10-15 07:28:37,136 [shard 0:comp] compaction - [Compact system_schema.view_virtual_columns 74226bd0-6b2c-11ee-b854-42bd0f7ab697] Compacted 2 sstables to [/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/data/system_schema/view_virtual_columns-08843b6345dc3be29798a0418295cfaa/me-3ga8_0kro_3we0g2swuzk9y7jmaf-big-Data.db:level=0]. 11kB to 6kB (~55% of original) in 371ms = 32kB/s. ~256 total partitions merged to 7.
INFO  2023-10-15 07:28:37,190 [shard 0:comp] compaction - [Compact system_schema.types 7472aeb0-6b2c-11ee-b854-42bd0f7ab697] Compacting [/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/data/system_schema/types-5a8b1ca866023f77a0459273d308917a/me-3ga8_0krk_1espc2swuzk9y7jmaf-big-Data.db:level=0:origin=memtable,/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/data/system_schema/types-5a8b1ca866023f77a0459273d308917a/me-3ga8_0kre_1k5ls2swuzk9y7jmaf-big-Data.db:level=0:origin=memtable]
INFO  2023-10-15 07:28:38,034 [shard 0:comp] compaction - [Compact system_schema.types 7472aeb0-6b2c-11ee-b854-42bd0f7ab697] Compacted 2 sstables to [/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/data/system_schema/types-5a8b1ca866023f77a0459273d308917a/me-3ga8_0krp_1inlc2swuzk9y7jmaf-big-Data.db:level=0]. 11kB to 5kB (~52% of original) in 665ms = 16kB/s. ~256 total partitions merged to 2.
INFO  2023-10-15 07:28:38,070 [shard 0:goss] schema_tables - Altering ks.cf id=5cf5a6c0-6b2c-11ee-9e66-5fe7028648ff version=9d351374-8a53-305f-bebe-9221044c1bb9
INFO  2023-10-15 07:28:38,072 [shard 0:comp] compaction - [Compact system_schema.keyspaces 74fbb4d0-6b2c-11ee-b854-42bd0f7ab697] Compacting [/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/data/system_schema/keyspaces-abac5682dea631c5b535b3d6cffd0fb6/me-3ga8_0krj_2u0u82swuzk9y7jmaf-big-Data.db:level=0:origin=memtable,/jenkins/workspace/scylla-master/dtest-debug/scylla/.dtest/dtest-liyr46dr/test/node2/data/system_schema/keyspaces-abac5682dea631c5b535b3d6cffd0fb6/me-3ga8_0krf_01xg02swuzk9y7jmaf-big-Data.db:level=0:origin=memtable]
INFO  2023-10-15 07:28:38,077 [shard 0:goss] query_processor - Column definitions for ks.cf changed, invalidating related prepared statements
INFO  2023-10-15 07:28:38,079 [shard 1:goss] query_processor - Column definitions for ks.cf changed, invalidating related prepared statements
```

**Comment by bhalevy** [comment]
Is https://github.com/scylladb/scylladb/issues/15686#issuecomment-1755608750 specific to raft consistent topology changes or is it a generic issue?

Cc @piodul @kbr-scylla @tgrabiec

**Comment by bhalevy** [comment]
Apparently also https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-debug/258/testReport/schema_management_test/TestSchemaManagement/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split014___test_update_schema_while_node_is_killed_create_table_/
```
cassandra.InvalidRequest: Error from server: code=2200 [Invalid query] message="unconfigured table cf"
```

**Comment by bhalevy** [comment]
@avikivity @mykaul given the fallout breadth I think this issue's priority should be increased to P0

**Comment by avikivity** [comment]
Agree, done

**Comment by piodul** [comment]
> Is [#15686 (comment)](https://github.com/scylladb/scylladb/issues/15686#issuecomment-1755608750) specific to raft consistent topology changes or is it a generic issue?
> 
> Cc @piodul @kbr-scylla @tgrabiec

This is specific to consistent topology changes only.

**Comment by kbr-scylla** [comment]
@bhalevy the failures you posted are unrelated to the OP.

**Comment by kbr-scylla** [comment]
Therefore I don't think this issue should be P0.

**Comment by kbr-scylla** [comment]
original post

**Comment by bhalevy** [comment]
As long it gets fixed as soon as possible without compromising higher priority issues, I don't care what label we use :)

**Comment by gleb-cloudius** [comment]
> As long it gets fixed as soon as possible without compromising higher priority issues, I don't care what label we use :)

This issue here does not affect CI. It should be fixed before topology over raft goes to production which has nothing to do with CI stability.

**Comment by kbr-scylla** [comment]
@piodul I think the schema sync part is fixed now, please confirm.
The "failed to resolve IP" is a dup of https://github.com/scylladb/scylladb/issues/14803, so if the schema sync part is fixed, I'll close this issue.

**Comment by piodul** [comment]
> @piodul I think the schema sync part is fixed now, please confirm.

Yes, it was fixed by https://github.com/scylladb/scylladb/pull/15724.
